latch/type_engine/__init__.py

```diff
@@ -0,0 +1,393 @@
+        if literal_type.simple == 9:
+            return _guess_python_class(literal_type)
+        return _simple_table[literal_type.simple]
+
+    if literal_type.collection_type is not None:
+        return typing.List[guess_python_type(literal_type.collection_type)]
+
+    if literal_type.blob is not None:
+
+        dim = literal_type.blob.dimensionality
+        if dim == 0:
+            return LatchFile
+        else:
+            return LatchDir
+
+    if literal_type.union_type is not None:
+
+        variant_types = [
+            guess_python_type(variant) for variant in literal_type.union_type.variants
+        ]
+
+        unique_variants = []
+        for t in variant_types:
+            if t not in unique_variants:
+                unique_variants.append(t)
+
+        return typing.Union[tuple(unique_variants)]
+
+    if literal_type.enum_type is not None:
+
+        escaped_variant_names = list(
+            map(
+                lambda x: "".join(
+                    filter(str.isidentifier, x.lower().replace(" ", "_"))
+                ),
+                literal_type.enum_type.values,
+            )
+        )
+
+        # Construct a unique symbol to represent each enum as a python class.
+        global _enum_counter
+        python_enum = enum.Enum(
+            f"enum{_enum_counter}", {x: x for x in escaped_variant_names}
+        )
+        _enum_counter += 1
+
+        return python_enum
+
+    raise NotImplementedError(
+        f"The flyte literal_type {literal_type} cannot be transformed to a python type."
+    )
+
+
+def best_effort_python_val(t: typing.T):
+    """Produce a "best-effort" default value given a python type.
+
+    Unlike `guess_python_val` there is no Literal value acting as a guide.
+    """
+
+    if t in _primitive_table:
+        return _primitive_table[t]
+
+    if t is list:
+        return []
+
+    file_like_table = {
+        LatchDir: LatchDir("latch:///foobar"),
+        LatchFile: LatchFile("latch:///foobar"),
+    }
+    if t in file_like_table:
+        return file_like_table[t]
+
+    if type(t) is enum.EnumMeta:
+        return list(t.__members__.values())[0]
+
+    if get_origin(t) is list:
+        list_args = get_args(t)
+        if len(list_args) == 0:
+            return []
+        return [best_effort_python_val(arg) for arg in list_args]
+
+    if get_origin(t) is typing.Union:
+        return best_effort_python_val(get_args(t)[0])
+
+    if get_origin(t) is None:
+
+        if "__dataclass_fields__" in t.__dict__:
+            fields = t.__dict__["__dataclass_fields__"]
+            dataclass_args = {
+                k: best_effort_python_val(v.type) for k, v in fields.items()
+            }
+            return t(**dataclass_args)
+
+        # TODO : record implementation
+
+    raise NotImplementedError(
+        f"Unable to produce a best-effort value for the python type {t}"
+    )
+
+
+def guess_python_val(literal: _Literal, python_type: typing.T):
```

Comment:
```suggestion
def guess_python_val(literal: _Literal, python_type: typing.Type[object]):
```

Reply:
you should run `to_python_value` instead? I mean from normal lytekit type engine?